src/App.ts

```diff
@@ -366,19 +368,22 @@ export default class App {
       };
     }
 
-    // Initialize receiver
+    /* --------------------- Initialize receiver ---------------------- */
     if (receiver !== undefined) {
-      // Custom receiver
-      if (this.socketMode) {
-        throw new AppInitializationError('receiver cannot be passed when socketMode is set to true');
+      // Custom receiver supplied
+      if (this.socketMode === true) {
+        // socketMode = true should result in SocketModeReceiver being used as receiver
+        // TODO: Add case for when socketMode = true and receiver = SocketModeReceiver
+        // as this should not result in an error
+        throw new AppInitializationError('You cannot supply a custom receiver when socketMode is set to true.');
       }
       this.receiver = receiver;
-    } else if (this.socketMode) {
+    } else if (this.socketMode === true) {
       if (appToken === undefined) {
-        throw new AppInitializationError('You must provide an appToken when using Socket Mode');
+        throw new AppInitializationError('You must provide an appToken when socketMode is set to true. To generate an appToken see: https://api.slack.com/apis/connections/socket#token');
       }
       this.logger.debug('Initializing SocketModeReceiver');
-      // Create default SocketModeReceiver
+      // Set receiver to SocketModeReceiver
```

Comment:
I think the code this comment corresponds to is self-evident.

Reply:
I agree with @misscoded's suggestion.

```suggestion
```